sys/kern/vmem.c

```diff
@@ -270,12 +287,25 @@ int vmem_alloc(vmem_t *vm, vmem_size_t size, vmem_addr_t *addrp,
   WITH_MTX_LOCK (&vm->vm_lock) {
     vmem_check_sanity(vm);
 
+  retry:
     bt = bt_find_freeseg(vm, size);
 
     if (bt == NULL) {
-      pool_free(P_BT, btnew);
       klog("%s: block of %lu bytes not found in '%s'", __func__, size,
            vm->vm_name);
+      if (grow_called || !(vm->vm_flags & VM_GROW))
```

Comment:
I noticed that `kva_alloc()` passes the `M_NOGROW` flag to `vmem_alloc()`. First, I'm not sure `kva_alloc()` should to that. Second, we should do something with that flag here.

Reply:
I'm not sure what is a purpose of `M_NOGROW`. We don't use that flag in `vmem` code.
@cahirwpz could you explain origin of that flag?

Reply:
I have moved everything to `kmem`.